audit_event.go

```diff
@@ -0,0 +1,110 @@
+package tfe
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ AuditEvents = (*auditEvents)(nil)
+
+// AuditEvents describes all the audit event related methods that the
+// Terraform Cloud API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/audit-trails.html
+type AuditEvents interface {
+	// List audit events visible to the current user.
+	List(ctx context.Context, options AuditEventListOptions) (*AuditEventList, error)
+}
+
+// auditEvents implements AuditEvents.
+type auditEvents struct {
+	client *Client
+}
+
+// AuditEvent represents a Terraform Cloud audit event.
+type AuditEvent struct {
+	ID        string              `json:"id"`
+	Version   string              `json:"version"`
+	Type      string              `json:"type"`
+	Timestamp time.Time           `json:"timestamp,iso8601"`
+	Auth      *AuditEventAuth     `json:"auth"`
+	Request   *AuditEventRequest  `json:"request"`
+	Resource  *AuditEventResource `json:"resource"`
+}
+
+// AuditEventList represents a list of audit events.
+type AuditEventList struct {
+	Pagination struct {
+		CurrentPage  int `json:"current_page"`
+		PreviousPage int `json:"prev_page"`
+		NextPage     int `json:"next_page"`
+		TotalPages   int `json:"total_pages"`
+		TotalCount   int `json:"total_count"`
+	} `json:"pagination"`
```

Comment:
This is the ugly thing that had to be done in order for the pagination data to unmarshal correctly. Note the underscores in the JSON tag names (this reflects the raw response).